src/pepr/operator/controllers/network/generators/kubeAPI.ts

```diff
@@ -0,0 +1,46 @@
+import { V1NetworkPolicyPeer } from "@kubernetes/client-node";
+import { K8s, Log, kind } from "pepr";
+
+import { anywhere } from "./anywhere";
+
+let apiServerPeers: V1NetworkPolicyPeer[];
+
+void getAPIServerCIDR();
+
+/**
+ * This generates a NetworkPolicyPeer that matches the API server endpoints.
+ *
+ * @returns A NetworkPolicyPeer that matches the API server endpoints
+ */
+export function generateKubeAPI(): V1NetworkPolicyPeer[] {
+  return apiServerPeers;
+}
+
+async function getAPIServerCIDR(): Promise<V1NetworkPolicyPeer[]> {
+  try {
+    // @todo: evaluate if this ever changes with node autoscaling
+
+    // Read the API server endpoints from the cluster
+    const { endpoints } = await K8s(kind.EndpointSlice).InNamespace("default").Get("kubernetes");
```

Comment:
This will deny access to new EndpointSlices (new IPs) for the Kubernetes API, which I believe will happen if control plane nodes are replaced. Most things in-cluster will connect through the `kubernetes.default.cluster.local` Service which has a ClusterIP. These EndpointSlices are the external IPs of the k8s control plane nodes.

Not sure how to best handle this. Re-running the operator would fix it, but the operator will reconcile infrequently. I suppose you could have a watcher on EndpointSlice and update NetworkPolcies automatically, but that would also have a slight lag too (maybe that's acceptable, I think it's rare that things are connecting to the external IPs of the k8s API anyway?).

Reply:
There is a todo here to check this.

If you want a quick demo:

```
k3d cluster create
kubectl get endpointslice kubernetes -o yaml

k3d node  create --role server node-foo
kubectl get endpointslice kubernetes -o yaml
```

Reply:
@bburky is there a different way you've seen kube api egress done before (different place to grab the proper IP(s) for a network policy). Even though pods are connecting to `kubernetes.default.cluster.local` from what I've seen they need to have an egress policy to the endpointslices in order for that connection to work consistently.

Agreed on the issue with new nodes though (captured in the todo on line 21).

Reply:
Hmm, you're right. Rerhaps iptables in the container transforms the ClusterIP into the external one or something?

I was just testing locally and confrmed that `kubernetes.default.cluster.local` has a different IP and does not containe the individual EndpointSlices IPs (which is expected for non-headless Services). I tested creating a NetworkPolicy without the exact EndpointSlices and indeed the kubernetes ClusterIP becomes inaccessible.

Reply:
We could perhaps generate CNI specific policies, generate a CNI-specific allow rule, but I'm not sure if they get merged with native k8s NetworkPolicies. And doing CNI-specific things would be complicated.

https://docs.tigera.io/calico/latest/network-policy/policy-rules/service-policy#allow-access-to-the-kubernetes-api-for-a-specific-namespace

`kube-apiserver` https://docs.cilium.io/en/stable/security/policy/language/#entities-based

For a generic solution Watch() on endpoints and updating policies is all I can think of

Reply:
You can't use clusterip, netpols dont care about that, they only care about the pod ip, so we have to use endpoint slices, sadly.

Reply:
okay based on our discussion, I have implemented a watch on the endpointslices and now trigger updates to netpols on changes (if required)